### ISSUE TITLE
Content quality curation

### ISSUE THREAD

**Issue by Chalarangelo** [30-seconds/30-seconds-web#166]
In an effort to revitalize the website and ensure high quality, we are taking a look at various metrics and data. Here's some preliminary research on current user behavior based primarily on Google Analytics:

  | Pageviews | Avg. time on page | Entrances | Bounce rate | Exit rate | Snippets | Views/Snippet
-- | -- | -- | -- | -- | -- | -- | --
Blog | 25485 | 1:54 | 9251 | 78.01% | 47.11% | 57 | 447.11
JavaScript | 48103 | 1:04 | 8301 | 39.35% | 19.68% | 435 | 110.58
CSS | 24406 | 0:44 | 2486 | 42.00% | 14.34% | 62 | 393.65
Python | 12771 | 0:50 | 1954 | 39.23% | 20.34% | 116 | 110.09
React | 12176 | 1:03 | 2402 | 56.49% | 27.15% | 51 | 238.75
Dart | 1529 | 0:37 | 191 | 52.36% | 20.41% | 81 | 18.88
C# | 4284 | 0:42 | 704 | 47.52% | 20.96% | 80 | 53.55
PHP | 5375 | 0:44 | 648 | 37.81% | 16.17% | 48 | 111.98
Go | 2344 | 0:40 | 220 | 36.36% | 18.26% | 60 | 39.07

### Insights

- Projects such as **Go** and **Dart** are not all that successful, therefore we might want to cut our losses and unlist them (while still allowing them to be viewed and found on the website), so that they don't clutter any of the main UX and allow users to find content more relevant to them.
- The **PHP** project seems very successful, even if we have put very little effort into it so far, which is indicative that we should focus some of our efforts to increase its visibility, improve quality and add more content.
- The **C#** project is underperforming, but is not in such a bad place, so we should give it some more space and track its results.
- The **React** project has a very high bounce rated compared to other content, which indicates we should revamp it and focus on improving its quality and/or adding some more content.
- The **Python** project has a very high performance, we should focus some of our efforts to improve and add more content to it, as we can.
- The **CSS** project has a great performance, however average time on page is a bit lower than our average. We should focus some of our efforts on improving the content and adding more content to it.
- The **JavaScript** project is considered the baseline and its performance is very satisfactory. Maintenance, content improvement and new content are all wanted but not a priority right away.
- The **Blog** project is by far the highest value project of all, even though it sees high bounce rates due to search entrances etc. We definitely need to focus efforts here and add more content.

### Incentives

- Add an *unlisted* flag for projects, so that we can soft hide **Go** and **Dart**. We might also want to figure out a way to freeze rankings on these projects.
- Improve navigation for **Blog** by adding tags/languages at the top. #156 
- Work on **PHP**, releasing at least another 32 snippets to bring it up to par with **C#**.
- Create some blogs or relevant content for **C#** to test the waters and reach a final decision.
- Revamp all **JavaScript, CSS, React** and **Python** snippets to include SEO-friendly descriptions and a better style that is friendlier to our target audience.

**Comment by Chalarangelo** [comment]
### List formatting

It seems that list formatting is way easier to scan through. We are going to convert all repositories so that snippet descriptions are formatted like lists:

- [x] Dart
- [ ] Go
- [ ] PHP
- [ ] C#
- [ ] Python
- [ ] JavaScript

**Comment by Chalarangelo** [comment]
A mock design to spice up snippet pages with decorative imager could look something like this:

![image](https://user-images.githubusercontent.com/8281875/93220760-9d44fd80-f775-11ea-9d90-a6a35d1ea766.png)

This kind of layout could also apply to blog snippets with a little bit of additional experimentation.

**Comment by Chalarangelo** [comment]
### Hotjar data

We gathered some initial data to analyze behavior and figure out next steps for content and platform. Screenshots are samples, so they represent all matching snippets.

#### JS Snippet interactions

![heatmap-6565444-click-desktop](https://user-images.githubusercontent.com/8281875/93298367-c7410300-f7fb-11ea-95b9-88198bf174e9.jpg)


#### CSS Snippet interactions

![heatmap-6565445-click-desktop](https://user-images.githubusercontent.com/8281875/93298403-daec6980-f7fb-11ea-8e10-e10180f41027.jpg)

#### Main Listing scroll heatmap

![heatmap-6565441-scroll-desktop](https://user-images.githubusercontent.com/8281875/93298495-f6577480-f7fb-11ea-9cf0-53847ae88dd8.jpg)


---

### Results and next steps

- From the above data, it is clear that listings are too long and that the CTA could be higher on the page. We should experiment with shorter listing pages (30 or 20 items per page) as well as the CTA being earlier in the list (25% or 33% of the list's length).
- The CodePen button is significantly more used than the copy button. This indicates we might want to explore using something like Runkit or Repl.it for snippets in JS and other similar languages.
- Breadcrumb usage is higher than expected, the issue for updating them (#165) should be prioritized as soon as possible.
- Assumptions about language icons and tags needing to be clickable have not yet been verified, however we will move forward with these as soon as we get a chance.

**Comment by Chalarangelo** [comment]
### Pageviews data from Jan. 1st, 2020 to Sep. 29th, 2020


Page path level 1 | Pageviews | Unique Pageviews | Avg. Time on Page | Bounce Rate | % Exit
-- | -- | -- | -- | -- | --
/js/ | 162,709 | 123,460 | 0:00:54 | 35.94% | 14.99%
/css/ | 99,218 | 70,131 | 0:00:34 | 37.45% | 9.86%
/ | 63,042 | 39,205 | 0:00:30 | 25.37% | 23.13%
/blog/ | 42,935 | 39,094 | 0:01:47 | 75.88% | 41.45%
/https:/ | 40,169 | 37,026 | 0:00:56 | 31.53% | 23.91%
/react/ | 33,030 | 25,594 | 0:00:51 | 52.51% | 18.65%
/python/ | 32,913 | 25,436 | 0:00:45 | 37.50% | 16.82%
/list/ | 13,163 | 8,454 | 0:00:20 | 33.76% | 11.40%
/php/ | 12,328 | 9,556 | 0:00:38 | 34.59% | 13.32%
/c-sharp/ | 8,726 | 7,061 | 0:00:40 | 45.46% | 16.91%
  | 540,408 | 411,104 | 0:00:46 | 38.25% | 17.91%

**Comment by Chalarangelo** [comment]
The latest data suggests that PHP and C# might be better off as unlisted. Also, the Blog button on the main listing has very few clicks. We might want to investigate: 

- A new icon
- Position
- If all blogs could be categorized in other languages and unlist the blog listing altogether somehow.

**Comment by Chalarangelo** [comment]
Hidden C# and PHP from the front page. Also, Heatmap data gathering has finished. We noticed higher click rates to categories in general, as well as better scroll % for the home page. Another Heatmap for JS listing pages is still running.

**Comment by Chalarangelo** [comment]
The JS Heatmap results are in. Everything is at around the same level as the home page, with average fold being later in the page (includes the Twitter button) and users using most of the page's controls equally. 

An interesting observation is that the CTA in listings might be where the fold ends, so we should investigate if that is the cause or the result here.

Additionally, we are concluding the git-related Survey. Results can be found below:

![image](https://user-images.githubusercontent.com/8281875/95043588-e8ec2680-06e5-11eb-91c2-b2ac37265b66.png)


As you can see from the replies, git seems like a valid topic to add next, at least to most users. Due to this being the first topic researched this way, it will make for a very interesting case study of original expectations and actual performance of a new category. We might also be able to follow up with additional surveys to see what users think of the content we add.

**Comment by Chalarangelo** [comment]
Meanwhile, we are getting a ton of spam and low-quality PRs due to Hacktoberfest. We suspect the damage to relations with the community will not be major, as most of those PRs are hit-and-runs from people who do not contribute to open source except for Hacktoberfest. We might also want to employ more strict guidelines in the future, so we can more easily curate low value snippets, as the projects have grown very large and there is very little new stuff people can add to make them better.

**Comment by Trinityyi** [comment]
After a week-long curation of the JS content repo, all snippets have been updated. Here's a quick breakdown of the changes we made:

- Formatted code blocks to a max of 80 characters per line.
- Replaced generic descriptions like `Returns true if X, false otherwise` in favor of descriptive verbs such as `Calculates`, `Creates` and `Checks`.
- Shortened descriptions to be more SEO-friendly and easier for users to read and understand.
- Improved explanations to include more of the key points wherever necessary/possible.
- Updated tags to better reflect expertise ratings, taking into account all the content we have available.
- Use of `.prototype` and other types made more consistent with MDN's conventions.
- Improved tags, especially for multi-category snippets.
- Removed low-use tags and tags that were redundant.
- Improved code of snippets wherever necessary.

The expertise retagging had the result of shifting content a little bit. The results are (discrepancy in the numbers is due to 2 newly added snippets):

**Before:** 167 Beginner, 257 Intermediate, 39 Advanced
**After:** 164 Beginner, 249 Intermediate, 52 Advanced

🎉 🎉 🎉

**Comment by Trinityyi** [comment]
Another quick curation and all Python snippets are updated and more consistent. Same changes as the JS ones.

🎉 🎉 🎉

**Comment by Trinityyi** [comment]
React snippets have also been updated for consistency and SEO, same as JS and Python ones.

**Comment by Chalarangelo** [comment]
CSS snippets have been updated for consistency and SEO, same as all the others. Closing the issue as we are done with everything for the time being.
🎉 🎉 🎉
